src/train.py

```diff
@@ -0,0 +1,70 @@
+from typing import Dict
+import pandas as pd
+import os
+from joblib import dump
+from model import (
+    bag_of_words,
+    mlb,
+    tf_idf
+)
+from data.preprocessing import read_data
+
+
+def get_corpus_counts(X_train: list[str], y_train: list[str]):
+    tags_counts: Dict[str, int] = {}
+    words_counts: Dict[str, int] = {}
+
+    for sentence in X_train:
+        for word in sentence.split():
+            if word in words_counts:
+                words_counts[word] += 1
+            else:
+                words_counts[word] = 1
+
+    for tags in y_train:
+        for tag in tags:
+            if tag in tags_counts:
+                tags_counts[tag] += 1
+            else:
+                tags_counts[tag] = 1
+
+    return words_counts, tags_counts
+
+
+def main():
+    train = read_data('assets/data/processed/train.tsv')
+    validation = read_data('assets/data/processed/validation.tsv')
+    test = pd.read_csv('assets/data/processed/test.tsv', sep='\t')
+
+    X_train, y_train = train['title'].values, train['tags'].values
+    X_val, y_val = validation['title'].values, validation['tags'].values
+    X_test = test['title'].values
+
+    words_counts, tags_counts = get_corpus_counts(X_train, y_train)
+    X_train_mybag, X_val_mybag = \
+        bag_of_words.initialize(words_counts, X_train, X_val, X_test)
+
+    X_train_tfidf, X_val_tfidf, _, tfidf_vocab = \
+        tf_idf.tfidf_features(X_train, X_val, X_test)
+    tfidf_reversed_vocab = {i: word for word, i in tfidf_vocab.items()}
+
+    mlb_classifier, y_train, y_val = mlb.get_mlb(tags_counts, y_train, y_val)
+    classifier_mybag = mlb.train_classifier(X_train_mybag, y_train)
+    classifier_tfidf = mlb.train_classifier(X_train_tfidf, y_train)
+
+    y_val_predicted_labels_mybag = classifier_mybag.predict(X_val_mybag)
```

Comment:
Maybe perform validation run in `evaluate.py` since you also show the results from it there?

Reply:
Any ideas on how to do this in a clean way? Without refactoring the `tf_idf` and the `bag_of_words` functions I have to also dump the `X_val_mybag` and `X_val_tfidf` objects, and then load them in `evaluate.py` which just feels like a sideways step.

Reply:
Personally Id say we split `initialize()` bc now it does the same thing for 3 datasets and uses the X_test only for printing the shape (it doesnt return it). We can do a similar thing for `tfidf_features()`. Then you only have to `initialize`/`tfidf_features` the train datasets here, meanwhile you save the `tfidf_vectorizer` and the `index_to_words` to use during evaluation.

Reply:
What are your thoughts btw @adondera? If we fix this, then we can merge it and continue with the next PRs